app.js

```
@@ -0,0 +1,468 @@
+        );
+
+        fileResults.push({
+          file,
+          ...scanFileResult,
+          ...storeResult,
+        });
+      }
+
+      console.log('Finished scanning files.');
+      console.log('\nDetailed results per file:');
+      console.dir(fileResults, { depth: null });
+
+      console.log('\nFiles per STIX Malware Analysis result:');
+      const resultValues = [
+        ...new Set([
+          ...STIX_MALWARE_RESULT_OV,
+          ...fileResults
+            .map((fileResult) => fileResult.stixMalwareAnalysis.result)
+            .sort(),
+        ]),
+      ];
+      resultValues.map((resultValue) => {
+        console.log('- ' + resultValue + ' :');
+        console.log(
+          fileResults
+            .filter(
+              (fileResult) =>
+                fileResult.stixMalwareAnalysis.result === resultValue,
+            )
+            .map((fileResult) => fileResult.file),
+        );
+      });
+
+      // TODO: Let storeMalwareAnalysis() run a SELECT query after the insert
+      //       to check in which (if any) graphs the resource was inserted?
+      //       Would be more reliable and easier to parse.
+      const filesNoDatabaseUpdate = fileResults.filter(
+        (fileResult) =>
+          !(
+            fileResult.databaseResponse?.results?.bindings[0]?.['callret-0']
+              ?.value &&
+            fileResult.databaseResponse?.results?.bindings[0]?.[
+              'callret-0'
+            ]?.value.match(' -- done')
+          ),
+      );
+      if (filesNoDatabaseUpdate.length) {
+        console.log(
+          '\nFiles for which the database response indicates that the ' +
+            'malware analysis resource object was not added to any graph:',
+        );
+        console.dir(filesNoDatabaseUpdate, { depth: null });
+      }
+    } catch (e) {
+      console.log(e);
+      res.status(500).send('Uncaught error in /delta: ' + e);
+      // TODO: Re-throw error? Because not sure if the response 500
```

Here I think that's the good response, for now.
Currently this is a silent fail; meaning; no one will get notified if we have a general exception. And here is a bit a problem of convention; there is no general way of dealing with these kind of situations.
In loket apps for instance; it's gonna write an error to the database and that will be picked up by another service who will send a mail. But unsure what others do ATM.
Anyway; i would keep for this iteration the handling as is. Maybe use `console.error` instead

---

Btw: https://github.com/mu-semtech/mu-javascript-template#error-handling